decentraland/unity-explorer
# Feat: Add Playables to control Avatar (#1999)

## What does this PR change?

It adds custom features that let a Unity user create Timeline animations to move, rotate and teleport avatars, and also to make them play emotes.
It implements a track and several playables that run outside of the ECS world. In order to make this work, several dirty tricks had to be implemented (which will not ship to production).
https://github.com/decentraland/unity-explorer/issues/1999

## How to test the changes?

1. Launch the explorer, with Unity.
2. Open the Timeline editor window.
3. Create a Timeline asset by instantiating a GameObject in the scene, selecting it and pressing the Create button that will appear in the Timeline window (this can be done without playing too). Uncheck the Play On Awake option in the PlayableDirector component that has just been added to the GameObject.
4. In the Timeline window, add a new track of type DCL.Animation > Avatar track.
5. Using the debug panel, create a random avatar and locate it in the scene hierarchy.
6. Select again the GameObject you created for the Timeline, and drag the AvatarBase object into the empty field of the track.
7. Now right click on the clips area of the track, add any clip related to avatars and set them up.
8. Press the Play button of the Timeline window. The random avatar should animate according to the added clips.

https://github.com/user-attachments/assets/ef2c3937-1574-4e71-b6b7-e73708fa7342
